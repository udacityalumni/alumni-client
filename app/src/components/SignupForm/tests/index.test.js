import SignupForm from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { fields } from './mocks';

describe('<SignupForm />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <SignupForm {...fields} isValid onSubmit={(e) => e} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
