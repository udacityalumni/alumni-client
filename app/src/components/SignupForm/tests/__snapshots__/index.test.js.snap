exports[`<SignupForm /> should render with default props 1`] = `
<Box
  announce={false}
  direction="column"
  focusable={true}
  pad={
    Object {
      "horizontal": "large"
    }
  }
  responsive={true}
  tag="div">
  <Form
    compact={false}
    fill={false}
    pad="none">
    <Box
      align="center"
      announce={false}
      direction="column"
      focusable={true}
      justify="center"
      pad="none"
      responsive={true}
      tag="div">
      <img
        src="https://github.com/RyanCCollins/cdn/blob/master/alumni-webapp/udacity-alumni-png.png?raw=true"
        style={
          Object {
            "height": "auto",
            "maxWidth": 150
          }
        } />
    </Box>
    <Heading
      align="center"
      strong={true}
      tag="h1">
      Udacity Alumni
    </Heading>
    <Heading
      align="center"
      tag="h5">
      Signup
    </Heading>
    <FormFields>
      <FormField
        error="value Required"
        help="What should we call you?"
        htmlFor="nameInput"
        label="Name"
        size="medium"
        strong={false}>
        <input
          active={false}
          autofilled={false}
          dirty={false}
          error="value Required"
          id="nameInput"
          initialValue=""
          invalid={true}
          name="name"
          placeholder="Bill Richardson"
          pristine={true}
          touched={true}
          type="text"
          valid={false}
          value=""
          visited={true} />
      </FormField>
      <FormField
        error={null}
        help="How should we get in touch with you?"
        label="Email"
        size="medium"
        strong={false}>
        <input
          active={false}
          autofilled={false}
          dirty={false}
          error="value Required"
          id="emailInput"
          initialValue=""
          invalid={true}
          name="email"
          placeholder="me@udacity.com"
          pristine={true}
          touched={false}
          type="email"
          valid={false}
          value=""
          visited={false} />
      </FormField>
      <FormField
        error={null}
        help="Make it secure"
        label="Password"
        size="medium"
        strong={false}>
        <input
          active={false}
          autofilled={false}
          dirty={false}
          error="value Required"
          id="passwordInput"
          initialValue=""
          invalid={true}
          name="passwordInput"
          pristine={true}
          touched={false}
          type="password"
          valid={false}
          value=""
          visited={false} />
      </FormField>
      <FormField
        error={null}
        help="Confirm your password"
        label="Password Confirmation"
        size="medium"
        strong={false}>
        <input
          active={false}
          autofilled={false}
          dirty={false}
          error="value Required"
          id="passwordConfirmationInput"
          initialValue=""
          invalid={true}
          name="passwordConfirmationInput"
          pristine={true}
          touched={false}
          type="password"
          valid={false}
          value=""
          visited={false} />
      </FormField>
    </FormFields>
    <Footer
      align="center"
      direction="row"
      pad={
        Object {
          "vertical": "medium"
        }
      }
      responsive={false}>
      <Button
        fill={true}
        label="Submit"
        onClick={[Function onSubmit]}
        primary={true}
        type="button" />
    </Footer>
    <undefined
      link="/login"
      text="Already a member?" />
  </Form>
</Box>
`;
